### PR TITLE
Structure_oM: Add new surface properties

### LINKED ISSUE #1480: Structure_oM: Add Hollow core Surface property
#### Description:

As title states, add support for hollow core surface Properties to be applied to panels

### LINKED ISSUE #1481: Structure_oM: Add VoidedFlatSlab SurfaceProperty
#### Description:

Add new surface properties:

- OneDirectionalVoidedSlab
- BiDirectionalVoidedSlab

### PR DESCRIPTION
**Pull request by IsakNaslundBh**
### Issues addressed by this PR

Closes #1480 
Closes #1481 

Adds a series of new SurfacePropery classes.

- Voided: One and two directional
- Hollow core (with a first two common opening profile types)
- Topped slab, to enable for example a precast slab with concrete topping.

### Test files

https://burohappold.sharepoint.com/:f:/s/BHoM/EnlCAJUsxb5Ckyaz48Avm1gBomY2EBiRpl1Fy8Vemwwozg?e=YoFIFQ

### CODE REVIEW

**Comment by IsakNaslundBh** [comment]
@BHoMBot check compliance

File: Structure_oM/SurfaceProperties/OneDirectionalVoided.cs [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+        [Description("A unique Name is required for some structural packages to create and identify the object.")]
+        public override string Name { get; set; }
+
+        [Length]
+        [Description("The thickness of the slab sitting on top of the ribs.")]
+        public virtual double TopThickness { get; set; }
+
+        [Length]
+        [Description("The thickness of the slab sitting beneth the ribs.")]
```
Comment (Chrisshort92) [inline Structure_oM/SurfaceProperties/OneDirectionalVoided.cs:45]: "beneath"

File: Structure_oM/SurfaceProperties/OneDirectionalVoided.cs [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        public virtual double BottomThickness { get; set; }
+
+        [Description("Homogenous structural material throughout the full thickness of the element.")]
+        public virtual IMaterialFragment Material { get; set; }
+
+        [Description("Specifies if the ribs are running in local x or y direction.")]
+        public virtual PanelDirection Direction { get; set; } = PanelDirection.X;
+
+        [Length]
+        [Description("Total depth measured from the bottom of the ribs to the top of the slab.")]
```
Comment (Chrisshort92) [inline Structure_oM/SurfaceProperties/OneDirectionalVoided.cs:55]: "Total depth measured from the bottom of the lower slab to the top of the upper slab"? The statement here implies the total depth is measured from the bottom of the ribs which would omit the bottom thickness.

**Review by Chrisshort92 — COMMENTED** [review]
Functionality of the PR has been tested with the attached test script. The test script performs as expected, correctly calculating volumes, densities, thicknesses, void ratios and material properties.

The code has been reviewed to ensure the correct engineering terminology has been adopted and that mathematical equations are correct (such as the void ratio). From this perspective, the code passes.

File: Structure_oM/SurfaceProperties/OneDirectionalVoided.cs (hunk 2 above)
Comment (IsakNaslundBh) [reply Structure_oM/SurfaceProperties/OneDirectionalVoided.cs:55]: Good catch, copy paste mistake on my end from RIbbed property. Will update to what you suggested!

**Review by Chrisshort92 — COMMENTED** [review]
Requested changes in the code ok

**Review by Chrisshort92 — APPROVED** [review]
Approved based on comments and changes above.

**Comment by IsakNaslundBh** [comment]
@BHoMBot check compliance

**Comment by IsakNaslundBh** [comment]
@BHoMBot check core
@BHoMBot check serialisation

**Comment by IsakNaslundBh** [comment]
@BHoMBot check ready-to-merge

**Comment by IsakNaslundBh** [comment]
@BHoMBot check ready-to-merge

**Pull request merged**